server/src/main/java/org/elasticsearch/cluster/HeapUsage.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.cluster;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * Record representing the heap usage for a single cluster node
+ */
+public record HeapUsage(String nodeId, String nodeName, long totalBytes, long freeBytes) implements ToXContentFragment, Writeable {
```

Comment:
I guess the `freeBytes` is probably modelled after `DiskUsage`. But I feel `usedBytes` (and in turn usedPercent) is more suitable here since that is what we actually measure and more commonly talked about? I think `DiskUsage` uses `freeBytes` due to the methods available via Java NIO's `FileStore` class.

Reply:
Done in 887bcafa0c2daf2f5be59d564755250ee4e99b70